upload-api/ucan-invocation.js

```diff
@@ -24,8 +37,56 @@ export async function parseUcanInvocationRequest (request) {
   const car = await CAR.codec.decode(bytes)
   const carCid = car.roots[0].cid.toString()
 
+  // @ts-expect-error 'ByteView<unknown>' is not assignable to parameter of type 'ByteView<UCAN<Capabilities>>'
+  const dagUcan = UCAN.decode(car.roots[0].bytes)
+
   return {
     bytes,
-    carCid
+    carCid,
+    value: {
+      // Workaround for:
+      // https://github.com/web3-storage/ucanto/issues/171
+      // https://github.com/multiformats/js-multiformats/issues/228
+      // @ts-ignore missing types
+      att: dagUcan.att.map(replaceAllLinkValues),
+      aud: dagUcan.aud.did(),
+      iss: dagUcan.iss.did()
+    }
+  }
+}
+
+/**
+ * @param {any} value
+ */
+export const replaceAllLinkValues = (value) => {
```

Comment:
Looked into npm modules available, but could not get something useful for it. If there are suggestions of modules you know that could help please let me know. 

This small function goes through the Object through all its depth and replaces all `Link` instances by a JSON based on proposal https://github.com/web3-storage/ucanto/issues/171 + https://github.com/multiformats/js-multiformats/issues/228

Once this lands in `multiformats`, we will be able to drop this adapter function.